client/src/containers/FormBuilderLoader/FormBuilderLoader.js

```diff
@@ -363,7 +365,11 @@ function mapStateToProps(state, ownProps) {
 
   const stateOverrides = schema && schema.stateOverride;
   const loading = schema && schema.metadata && schema.metadata.loading;
-  return { schema, submitting, values, stateOverrides, loading };
+  const changed = !!(
+    state.unsavedForms &&
+    state.unsavedForms.find(form => ownProps.identifier === form.name)
+  );
```

Comment:
Could we use the `isDirty()` API provided by redux-forms?
`isDirty(form.name, getFormState)(state);`